lib/ansible/playbook/conditional.py

```diff
@@ -67,123 +61,50 @@ def evaluate_conditional(self, templar: Templar, all_vars: dict[str, t.Any]) ->
-                if conditional is None or conditional == '':
-                    res = True
-                elif isinstance(conditional, bool):
-                    res = conditional
-                else:
+        for conditional in self.when:
+            if conditional is None or conditional == "":
+                res = True
+            elif isinstance(conditional, bool):
+                res = conditional
+            else:
+                try:
                     res = self._check_conditional(conditional, templar, all_vars)
+                except AnsibleError as e:
+                    raise AnsibleError(
+                        "The conditional check '%s' failed. The error was: %s" % (to_native(conditional), to_native(e)),
+                        obj=getattr(self, '_ds', None)
+                    )
 
-                # only update if still true, preserve false
-                if result:
-                    result = res
-
-                display.debug("Evaluated conditional (%s): %s" % (conditional, res))
-                if not result:
-                    false_condition = conditional
-                    break
-
-        except Exception as e:
-            raise AnsibleError("The conditional check '%s' failed. The error was: %s" % (to_native(conditional), to_native(e)), obj=ds)
+            display.debug("Evaluated conditional (%s): %s" % (conditional, res))
+            if not res:
+                return res, conditional
 
-        return result, false_condition
-
-    def _check_conditional(self, conditional, templar, all_vars):
-        '''
-        This method does the low-level evaluation of each conditional
-        set on this object, using jinja2 to wrap the conditionals for
-        evaluation.
-        '''
+        return True, None
 
+    def _check_conditional(self, conditional: str, templar: Templar, all_vars: dict[str, t.Any]) -> bool:
         original = conditional
-
-        if templar.is_template(conditional):
-            display.warning('conditional statements should not include jinja2 '
-                            'templating delimiters such as {{ }} or {%% %%}. '
-                            'Found: %s' % conditional)
-
-        # make sure the templar is using the variables specified with this method
         templar.available_variables = all_vars
-
         try:
-            # if the conditional is "unsafe", disable lookups
-            disable_lookups = hasattr(conditional, '__UNSAFE__')
-            conditional = templar.template(conditional, disable_lookups=disable_lookups)
+            if templar.is_template(conditional):
+                display.warning(
+                    "conditional statements should not include jinja2 "
+                    "templating delimiters such as {{ }} or {%% %%}. "
+                    "Found: %s" % conditional
+                )
+                conditional = templar.template(conditional)
+                if isinstance(conditional, bool):
```

Comment:
we could probably 'outdent' this if/else as it would also apply to untemplated conditional

Reply:
That wouldn't get here, it is already handled above in `evaluate_conditional_with_result`.

Reply:
tempted to move both into just the post_validate_X_when: functions